### ISSUE TITLE
[Feature Request] Support scopes and network limitations for API tokens

### ISSUE THREAD

**Issue by arbales** [zerotier/ZeroTierOne#2319]
It'd be useful for ZeroTier to support multiple scopes, so that services like `zeronsd` can have read-only access required to run, but not make other changes to one's network(s). Readonly, and per-network access would both be useful places to start.

**Comment by janjaapbos** [comment]
I have not checked, but I assume API tokens are created for a specific user account. You can eat those rights on a user account.

**Comment by arbales** [comment]
Yes, but they are read-write and apply to all networks. Read-only and network-scoped keys seem like a straightforward ask, and a tablestakes security feature for a product like ZeroTier.
